pennylane/operation.py

```diff
@@ -232,6 +232,92 @@ def expand_matrix(base_matrix, wires, wire_order=None):
     return mat
 
 
+def _local_sparse_swap_mat(i, n, format="csr"):
+    """Helper function which generates the sparse matrix of SWAP
+    for qubits: i <--> i+1 with final shape (2**n, 2**n)."""
+    assert i < n - 1
+    swap_mat = csr_matrix([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]])
+
+    j = i + 1  # i is the index of the qubit, j is the number of qubits prior to and include qubit i
+    return kron(
+        kron(eye(2 ** (j - 1)), swap_mat), eye(2 ** (n - (j + 1))), format=format
+    )  # (j - 1) + 2 + (n - (j+1)) = n
+
+
+def _sparse_swap_mat(i, j, n, format="csr"):
+    """Helper function which generates the sparse matrix of SWAP
+    for qubits: i <--> j with final shape (2**n, 2**n)."""
+    assert i < n and j < n
+    if i == j:
+        return eye(2**n, format=format)
+
+    (small_i, big_j) = (i, j) if i < j else (j, i)
+    store_swaps = [
+        _local_sparse_swap_mat(index, n, format=format) for index in range(small_i, big_j)
+    ]
+
+    res = eye(2**n, format=format)
+    for mat in store_swaps:  # swap i --> j
+        res @= mat
+
+    for mat in store_swaps[-2::-1]:  # bring j --> old_i
+        res @= mat
+
+    return res
+
+
+def sparse_expand_matrix(base_matrix, wires, wire_order=None, format="csr"):
```

Comment:
This is a good question 🤔, I know that the existing expand method keeps `wire_order` as a key word argument. It makes sense that it should be a mandatory argument otherwise what would be the purpose of such a function? Might be worth discussing if this is a change we can propagate with out too much refactoring in the code.

Reply:
For now we stick to the default implementation of `expand_matrix()`